docs/user_guide/configuring.rst

```diff
@@ -277,6 +277,97 @@ at the bottom. You can hide these buttons with the following configuration:
    }
 
 
+Add a dropdown to switch between docs versions
+==============================================
+
+You can add a button to your site's navbar or sidebars that allows users to
+switch between versions of the documentation.
+The links in the version switcher will differ depending on which page of the
+docs is being viewed. For example, on the page
+``https://mysite.org/en/v2.0/changelog.html``, the switcher links will go to
+``changelog.html``. When clicked, the switcher will check for the existence of that page, and if it doesn't exist, redirect to the homepage of that doc version instead.
+
+This is accomplished via the following configuration keys:
+
+- A JSON file that is a list the documentation versions that should be available on each page.
+- A ``json_url`` configuration that defines the persistent location of the file above, so that documentation from older docs versions have their switcher options defined by this single source.
+- A ``template_url`` configuration that defines how your documentation versions should be inserted into URLs to create links to other versions of your documentation.
+- A ``version_match`` configuration that tells the switcher the version of the documentation that is currently being browsed.
+
+Below is a more in-depth description of each of these configuration options, and steps to add a switcher to your documentation.
+
+Add a JSON file to define your switcher's versions
+--------------------------------------------------
+
+First, you must define a JSON file that acts as the single source of truth for what versions should be available to switch between. This file contains a list of entries that have two fields:
+
+- ``version``: a version number, as defined by the ``version_match`` key (see below)
+- ``name``: an optional name to display in the dropdown, instead of the version number. (e.g., "latest", "stable", "dev", etc).
+
+**You must include this file with your documentation's build**. For example, by adding it to a folder that is listed under your site's ``html_static_path`` configuration (see `the Sphinx static path documentation <https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path>`_ for more information.
```

Comment:
@choldgraf I don't think the "you must include..." statement is true.  It should work just fine to keep the JSON outside of the doc build tree (indeed, outside the repo altogether) as long as it lives at a stable URL (ideally on the same domain, to avoid CORS problems).  That approach requires maintainers who are willing to add new versions manually as part of their release process, but it seems like a use case that might be preferred by some projects?

Reply:
good point - maybe we should just delete that little warning since it is covered later on?

Reply:
> good point - maybe we should just delete that little warning since it is covered later on?

done.